static/app/views/explore/hooks/useAnalytics.tsx

```diff
@@ -0,0 +1,53 @@
+import {useEffect, useMemo} from 'react';
+
+import type {Organization} from 'sentry/types/organization';
+import {trackAnalytics} from 'sentry/utils/analytics';
+import type {useSpansQuery} from 'sentry/views/insights/common/queries/useSpansQuery';
+
+import type {Visualize} from './useVisualizes';
+
+export function useAnalytics({
+  result,
+  resultsMode,
+  visualizes,
+  organization,
+  groupBys,
+  userQuery,
+}: {
+  groupBys: string[];
```

Comment:
Perhaps we should call this `columns` and pass the group bys in aggregate mode and fields in samples mode.

Reply:
Yup good idea @Zylphrex